### PR TITLE
feat: add open-iscsi system extension

### PR DESCRIPTION
**Pull request by andrewrynhard**
This system extension enables open-iscsi
    
Components:
    
* `iscsid` running as an extension service
* `tgtd` running as an extension service
    
Signed-off-by: Andrew Rynhard <andrew@rynhard.io>
Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>
Signed-off-by: Noel Georgi <git@frezbo.dev>

### CODE REVIEW

File: iscsid/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# syntax=docker/dockerfile:1.3
+
+FROM scratch AS iscsid
+COPY --from=ghcr.io/talos-systems/kmod:v0.9.0 / /rootfs/usr/local/lib/containers
```
Comment (smira) [inline iscsid/Dockerfile:4]: I feel `v0.9.0` should be an `ARG PKGS` so that we can easily tweak that

File: iscsid/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+COPY --from=ghcr.io/talos-systems/libressl:v0.9.0 / /rootfs/usr/local/lib/containers
+COPY --from=ghcr.io/talos-systems/musl:v0.9.0 / /rootfs/usr/local/lib/containers
+COPY --from=ghcr.io/talos-systems/open-isns:v0.9.0 / /rootfs/usr/local/lib/containers
+COPY --from=ghcr.io/talos-systems/util-linux:v0.9.0 / /rootfs/usr/local/lib/containers
+COPY --from=ghcr.io/talos-systems/open-iscsi:v0.9.0 / /rootfs/usr/local/lib/containers
+COPY iscsid.yaml /rootfs/usr/local/etc/containers/iscsid.yaml
+COPY manifest.yaml /manifest.yaml
```
Comment (smira) [inline iscsid/Dockerfile:11]: nit: `\n`

**Review by smira — APPROVED** [review]

**Comment by anthr76** [comment]
Would multipathd also be required as well? Trying to get talos to work with the pure-storage csi

File: longhorn/tgtadm/pkg.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    env:
+      PREFIX: /usr/local
+    prepare:
+      - |
+        tar -xzf tgtadm.tar.gz --strip-components=1
+      - |
+        patch -p1 < /pkg/musl.patch
+    build:
+      - |
+        make
```
Comment (frezbo) [inline longhorn/tgtadm/pkg.yaml:21]: no nproc?

Comment (smira) [reply longhorn/tgtadm/pkg.yaml:21]: good point, I can add that, this was pretty much the first time I had it building lol

File: Makefile [hunk 4]
```diff
@@ -30,7 +30,7 @@ COMMON_ARGS += --build-arg=TAG=$(TAG)
 empty :=
 space = $(empty) $(empty)
 
-TARGETS = amd-ucode bnx2-bnx2x gvisor hello-world-service intel-ucode
+TARGETS = amd-ucode bnx2-bnx2x gvisor hello-world-service intel-ucode iscsi-tools longhorn-stubs
```
Comment (smira) [inline Makefile:33]: I think we should add new system extensions to `README.md` catalog

File: longhorn-stubs/manifest.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+version: v1alpha1
+metadata:
+  name: longhorn-stubs
+  version: v1.2.4
```
Comment (smira) [inline longhorn-stubs/manifest.yaml:4]: should use `$VERSION` ? I used the Longhorn version here

**Comment by frezbo** [comment]
https://www.cyberciti.biz/tips/howto-setup-linux-iscsi-target-sanwith-tgt.html tested by creating a target as mentioned here

**Review by smira — APPROVED** [review]

**Pull request merged**